.github/workflows/pull-request.yaml

```diff
@@ -15,6 +15,41 @@ jobs:
           python-version: '3.8'
           cache: 'pip'
       - uses: pre-commit/action@v3.0.0
+  unit-tests:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        # conda profile activation requires a login shell
+        shell: bash -el {0}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          submodules: recursive
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: opscenter-test
+          environment-file: test/environment.yml
```

Comment:
why not use app/ui/environment.yml?

Reply:
and a `conda install pytest`

Reply:
Well, I guess two reasons

1. `streamlit` 1.16.0 is the latest available version in snowflake's anaconda channel (can probably work around that without too much pain)
2. `snowflake-native-apps-permission` also appears to be missing from snowflake's channel which makes no sense to me

I'm not sure why the permissions SDK package wouldn't be on the public-facing anaconda repo.

Reply:
Yeah, i'm not sure here. I can't find any mention of the permissions SDK on the public internet. The mismatch of information_schema.packages and the snowflake conda repo (1.22.0 the latest on the former, 1.16.0 on the latter) is another annoying wrinkle.

I'm a little worried about changing the "real" environment.yml and accidentally breaking OpsCenter releases. Since the set of dependencies is effectively disjoint (what we think is being used at runtime vs. what is found on snowflake's conda channel), what do you think of having separate environment.yml files for "product" and "test"? Maybe I've missed some other way forward?

I can also try reaching out to our Snowflake contacts and see if they can shed any light on this.

Reply:
The environment file is separate from the channel being used isn't it? Why do we have to use the public snowflake anaconda channel to use conda? Maybe I missed something?

We already have a poetry definition of the all and it's dependencies and a conda one. I don't really want to add a 3rd to maintain.

Reply:
We don't have to use Snowflake's conda channel, but I thought that was a goal behind switching to Conda (because we assume their Conda channel better matches what we'd get at runtime running inside Snowflake). Agree that if we don't care about using the snowflake channel, this becomes more straightforward.

Pyproject.toml file does not list the native app permissions sdk as a dependency which is why we currently don't have this problem.

I can appreciate not wanting more files declaring conflicting lists of dependencies. However, I'm not sure how we could use one environment.yml file for both inside and outside of Snowflake. I guess we could run some bash commands in the GitHub workflow to delete the permissions dependency line before we run unit tests? I couldn't find anything like an optional dependency feature with Conda which I expected to be able to do.

Reply:
For this change I think we have two options. Encode this into the pyproject.toml or figure out a better solution for conda. I am inclined to lean towards pyproject.toml until we have a better understandign of how snowflake is going to do its anaconda channel.

I def don't want to mix anaconda and poetry for our internal tests though...

Reply:
Sounds good. I'll back out the conda changes, use poetry for this PR, and we'll take on the conda migration in #36